language_detector/main.py

```diff
@@ -1,9 +1,27 @@
+# -*- coding: utf-8 -*-
+
 """This is the entry point of the program."""
 
 from .languages import LANGUAGES
 
-
 def detect_language(text, languages=LANGUAGES):
-    """Returns the detected language of given text."""
-    # implement your solution here
-    pass
+    """Returns the detected lang of given text."""
+    numOfTimesLanguageOccurs = {}
+
+    for word in text.split():
+        lang = whichLanguageIsIt(word, languages)
+        if lang is not None:
+            addToDictionary(lang, numOfTimesLanguageOccurs)
+    return getHighestOccurringLanguage(numOfTimesLanguageOccurs)
```

Comment:
Instead of calling a function to create the return value I suggest:

    return max(numOfTimesLanguageOccurs.keys(), key = lambda key: numOfTimesLanguageOccurs[key])

This returns your answer without the need of another function call.